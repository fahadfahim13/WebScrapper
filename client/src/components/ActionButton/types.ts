import { ReactElement } from "react";
import {SvgIconProps} from "@mui/material";

import {BUTTON_COLORS, BUTTON_TYPES} from "shared/types";

export interface Props {
    variant?: `${BUTTON_TYPES}` | undefined;
    startIcon?: ReactElement<SvgIconProps>;
    value: string;
    color?: `${ BUTTON_COLORS }` | undefined;
    onClick: () => void;
}
