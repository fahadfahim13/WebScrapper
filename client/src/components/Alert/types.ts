import { SyntheticEvent, MouseEvent } from "react";
import { VERTICAL, HORIZONTAL } from "shared/types";

export interface Props {
    open: boolean;
    autoHideDuration?: number;
    message: string;
    vertical?: `${VERTICAL}`;
    horizontal?: `${HORIZONTAL}`;
    onClose: (event: SyntheticEvent | MouseEvent) => void;
}