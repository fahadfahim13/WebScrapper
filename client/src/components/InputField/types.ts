import { ChangeEvent } from "react";
import {InputTypes} from "store/FormInputs/types";

export interface Props {
    id?: string;
    name: InputTypes;
    value: string;
    type?: string;
    required?: boolean;
    label?: string;
    defaultValue?: string;
    onChange: (e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => void;
    multiline?: boolean;
    maxRows?: number;
    width?: string;
}