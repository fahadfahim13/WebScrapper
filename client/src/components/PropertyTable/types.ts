import { PropertyDetails } from "shared/types";

export interface Props {
    tableBody?: PropertyDetails[];
}