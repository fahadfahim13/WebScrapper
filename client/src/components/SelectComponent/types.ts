import { SelectChangeEvent } from "@mui/material";

export interface Option{
    id: number;
    value: string;
    text: string;
}

export interface Props {
    options: Option[];
    handleChange: (e: SelectChangeEvent) => void;
    value: Option;
    label?: string;
}