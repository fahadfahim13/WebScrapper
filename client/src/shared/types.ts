export enum BUTTON_TYPES {
    OUTLINED = "outlined",
    TEXT = "text",
    CONTAINED = "contained",
}

export enum BUTTON_COLORS {
    INHERIT = "inherit",
    PRIMARY = "primary",
    SECONDARY = "secondary",
    SUCCESS = "success",
    ERROR =  "error",
    INFO = "info",
    WARNING = "warning"
}

export enum VERTICAL {
    TOP = "top",
    BOTTOM="bottom",
}

export enum HORIZONTAL {
    CENTER = "center",
    LEFT = "left",
    RIGHT = "right",
}