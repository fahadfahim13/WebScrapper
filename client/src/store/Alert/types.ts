export interface State {
  id: string | number;
  message: string;
  open: boolean;
}
