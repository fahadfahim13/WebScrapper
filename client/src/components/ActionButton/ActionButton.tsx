import React from 'react';

import {Button} from '@mui/material';
import {Props} from "./types";

const ActionButton = (props: Props) => {
    const { variant="outlined", startIcon=null, value="Button", color="primary", onClick } = props;
    return (
        <Button variant={variant} startIcon={startIcon} color={color} onClick={onClick}>
            {value}
        </Button>
    );
};

export default ActionButton;