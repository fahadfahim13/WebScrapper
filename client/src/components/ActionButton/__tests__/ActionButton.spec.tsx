import {render, RenderResult} from '@testing-library/react';
import ActionButton from 'components/ActionButton';
import {Props} from '../types';
import {BUTTON_COLORS, BUTTON_TYPES} from 'shared/types';

let documentBody: RenderResult;

describe("<ActionButton />", () => {
    const props: Props = {
        variant: BUTTON_TYPES.OUTLINED,
        value: 'SAVE',
        color: BUTTON_COLORS.PRIMARY,
        onClick: () => {
            console.log("Clicked")
        },
    }
    beforeEach(() => {
        documentBody = render(<ActionButton {...props} />)
    });
    it('matches snapshot', () => {
        const {baseElement} = documentBody;
        expect(baseElement).toMatchSnapshot();
    })

});