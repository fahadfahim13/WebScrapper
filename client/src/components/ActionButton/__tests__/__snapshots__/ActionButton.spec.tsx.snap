// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionButton /> matches snapshot 1`] = `
<body>
  <div>
    <button
      class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      SAVE
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
